### PR TITLE
packetbeat protocols `enabled` config

### PR DESCRIPTION
**Pull request by urso**
blocked by #1987

### CODE REVIEW

**Comment by monicasarbu** [comment]
LGTM, waiting for green

**Comment by urso** [comment]
some additional changes next to adding `enable` flag to every protocol plugin:
- add `enable` flag to flows
- renamed `enabled` flag to `enable` in ICMP protocol plugin + change default to true (as config says default is false, but sets enabled to true). The `enabled` flag was not used in code, but `enable` flag is.

File: packetbeat/config/config.go [hunk 1]
```diff
@@ -35,8 +35,9 @@ type InterfacesConfig struct {
 }
 
 type Flows struct {
-	Timeout string
-	Period  string
+	Enable  bool   `config:"enable"`
```
Comment (urso) [inline packetbeat/config/config.go]: just discovered the default value being `false`, but not `true` as documented.

Comment (urso) [inline packetbeat/config/config.go]: fixed by introducing (*Flows).Enabled().

**Comment by ruflin** [comment]
@urso Same problem here as with the output PR. You used `enable`, Metricbeat uses `enabled`.

**Comment by urso** [comment]
@ruflin will change metricbeat to `enable` ;)

**Comment by ruflin** [comment]
Both works for me. I tried to find some example for `enable` or `enabled` in other projects. For example elasticearch (at least the tests) seem to use `enabled`: https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/test/search/60_query_string.yaml#L10 Most other projects I found also use `enabled`, but couldn't find "hard facts" for on one or the other.

**Comment by urso** [comment]
changed `(*common.Config).Enabled` method to look for `enabled` config setting. Did require to update docs for output modules too.

File: packetbeat/docs/reference/configuration/packetbeat-options.asciidoc [hunk 2]
```diff
@@ -247,6 +254,12 @@ packetbeat.protocols.thrift:
 
 The following options are available for all protocols:
 
+===== enabled
+
+The enabled setting is a boolean setting to enable or disable protocols without having to comment out configuration sections.. If set to false, the protocol is disabled.
```
Comment (ruflin) [inline packetbeat/docs/reference/configuration/packetbeat-options.asciidoc]: `sections..`

File: packetbeat/etc/beat.full.yml [hunk 3]
```diff
@@ -45,20 +45,27 @@ packetbeat.interfaces.device: any
 
 #================================== Flows =====================================
 
-# Set network flow timeout. Flow is killed if no packet is received before being
-# timed out.
-packetbeat.flows.timeout: 30s
+packetbeat.flows:
+  # Enable Network flows. Default: true
+  #enable: true
```
Comment (ruflin) [inline packetbeat/etc/beat.full.yml]: That should be enabled.

File: packetbeat/etc/beat.full.yml [hunk 4]
```diff
@@ -45,20 +45,27 @@ packetbeat.interfaces.device: any
...
+  # Configure reporting period. If set to -1, only killed flows will be reported
+  period: 10s
 
 #========================== Transaction protocols =============================
 
 packetbeat.protocols.icmp:
-  # Enable ICMPv4 and ICMPv6 monitoring. Default: false
-  enabled: true
+  # Enable ICMPv4 and ICMPv6 monitoring. Default: true
+  #enable: true
```
Comment (ruflin) [inline packetbeat/etc/beat.full.yml]: enabled

File: packetbeat/etc/beat.full.yml [hunk 5]
```diff
@@ -45,20 +45,27 @@ packetbeat.interfaces.device: any
...
 
 packetbeat.protocols.amqp:
+  # Enable AMQP monitoring. Default: true
+  #enable: true
```
Comment (ruflin) [inline packetbeat/etc/beat.full.yml]: enabled

File: packetbeat/etc/beat.full.yml [hunk 6]
```diff
@@ -93,6 +100,9 @@ packetbeat.protocols.amqp:
   #transaction_timeout: 10s
 
 packetbeat.protocols.dns:
+  # Enable DNS monitoring. Default: true
+  #enable: true
```
Comment (ruflin) [inline packetbeat/etc/beat.full.yml]: see all following

File: packetbeat/etc/beat.yml [hunk 7]
```diff
@@ -15,14 +15,14 @@ packetbeat.interfaces.device: any
...
 
-# Comment out all options to disable flows reporting.
-
-# Set network flow timeout. Flow is killed if no packet is received before being
-# timed out.
-packetbeat.flows.timeout: 30s
-
-# Configure reporting period. If set to -1, only killed flows will be reported
-packetbeat.flows.period: 10s
+# Set `enable: false` or comment out all options to disable flows reporting.
```
Comment (ruflin) [inline packetbeat/etc/beat.yml]: enabled

File: packetbeat/docs/reference/configuration/packetbeat-options.asciidoc [hunk 8]
```diff
@@ -283,9 +296,9 @@ The per protocol transaction timeout. Expired transactions will no longer be cor
 
 You can specify the following options in the `icmp` section of the +{beatname_lc}.yml+ config file:
 
-===== enabled
+===== enable
```
Comment (ruflin) [inline packetbeat/docs/reference/configuration/packetbeat-options.asciidoc]: enabled

Comment (urso) [inline packetbeat/docs/reference/configuration/packetbeat-options.asciidoc]: No idea why this still shows as enabled. Locally it's `enabled`, only opening the github editor for file in feature branch shows `enabled`. Some github caching going on here?

**Comment by ruflin** [comment]
@urso There is also an other issue with Enabled Method and variable: https://travis-ci.org/elastic/beats/jobs/144204760#L429

**Comment by urso** [comment]
@ruflin rebased and fixed build

**Pull request merged**